analysis/quantifier-comparisons/pre-processing-benchmarking-metrics.Rmd

```diff
@@ -0,0 +1,612 @@
+
+cellranger_feature_qc <- cellranger_sces %>%
+  purrr::map_df(genedata_to_df, .id = "quant_id")
+
+
+feature_qc <- dplyr::bind_rows(
+  alevin = alevin_feature_qc,
+  alevin_fry = alevin_fry_feature_qc, 
+  alevin_fry_sketch = alevin_fry_sketch_feature_qc,
+  kallisto = kallisto_feature_qc, 
+  cellranger = cellranger_feature_qc,
+  .id = "tool"
+) %>%
+  dplyr::left_join(quant_info,
+                   by = c("tool" = "tool", 
+                          "quant_id" = "quant_dir")) %>%
+  dplyr::filter(sample != "SCPCR000003")
+```
+
+```{r}
+gene_counts <- feature_qc %>% 
+  # remove genes that have a low frequency of being detected
+  dplyr::filter(detected >= 5.0) %>%
+  dplyr::count(gene_id, sample)
+
+common_genes <- gene_counts %>%
+  dplyr::filter(n == 5) %>%
+  dplyr::pull(gene_id)
+
+feature_qc_common <- feature_qc %>%
+  dplyr::filter(
+    (gene_id %in% common_genes) 
+  )
+```
+
+
+```{r}
+# make a table with information about each cell identified by a tool and how many other tools 
+# that cell is identified in
+gene_counts <- feature_qc %>%
+  dplyr::filter(detected >= 5.0) %>%
+  dplyr::select(tool, gene_id, mean, detected, sample) %>%
+  left_join(gene_counts)
+```
+
+
+```{r}
+gene_counts <- gene_counts %>%
+  group_by(tool, n) %>%
+  tally(name = "overlapping_genes") %>%
+  dplyr::rename("tools_detected" = "n")
+
+
+ggplot(gene_counts, aes(x = tools_detected, y = overlapping_genes, fill = tool)) + 
+  geom_col(position = "dodge") + 
+  theme_classic()
+```
+This plot isn't as informative as when looking at shared cells across each sample. In general it looks like the majority of the genes are identified across all samples, so let's actually look at a per cell level for this which might give us a much better picture. 
+
+
+```{r}
+# spread table for comparisons
+feature_compare <- feature_qc %>%
+  dplyr::filter(gene_id %in% detected_features) %>%
+  # spread the mean expression stats to one column per caller
+  tidyr::pivot_wider(id_cols = c(gene_id, sample),
+                     names_from = tool,
+                     values_from = mean) %>%
+  # drop rows with NA values to ease correlation calculations
+  tidyr::drop_na()
+```
+
+```{r}
+feature_compare %>% 
+  dplyr::group_by(sample) %>%
+  dplyr::summarize(
+    cr_al_cor = cor(cellranger, alevin, method = "spearman"),
+    cr_al_fry_cor =cor(cellranger, alevin_fry, method = "spearman"),
+    cr_al_fry_sketch_cor = cor(cellranger, alevin_fry_sketch, method = "spearman"),
+    cr_ka_cor = cor(cellranger, kallisto, method = "spearman")
+  )
+```
+So here we have correlation of mean expression/gene and the correlations here are much more varied than when looking at UMI/cell. We see that the highest correlation is between Alevin and Alevin-fry (without sketch), while Kallisto has the lowest correlation. It appears that the `--sketch` mode, although increases time (shown elsewhere), doesn't appear to have an increased advantage over Alevin-fry in terms of quantification. 
```

Comment:
> It appears that the `--sketch` mode, although increases time (shown elsewhere), doesn't appear to have an increased advantage over Alevin-fry in terms of quantification. 


I'd say this the opposite way... I would have expected `--sketch` to have a worse correlation, and it doesn't! Sketch mode seems good!